tests/erlang_tests/is_record.erl

```diff
@@ -0,0 +1,48 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Tomasz Sobkiewicz <tomasz.sobkiewt>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(is_record).
+
+-record(person, {id, name, age}).
+
+-export([start/0, id/1]).
+
+-define(ID(Arg), ?MODULE:id(Arg)).
+
+start() ->
```

Comment:
we are not testing errors here, such as when wrong type is used for argument 2: `erlang:is_record(..., 42)`.
Also we are not testing is_record when used as a guard (the compiler and the VM handle guards in a different way).